### PR TITLE
METRON-2060 Improving Alerts table config pane

### PR DESCRIPTION
**Pull request by tiborm**
## Contributor Comments

As a part of this PR I'm separating list visible in available columns in alerts table config pane. This makes a better user experience also fixes a few glitches.

List of changes:
- separating lists of visible and available columns 
- changing the width of the pane to make more space on the screen useful
- the layout of the pane automatically aligns to the width of the pane and the amount of content
- moving the lists beside each other (except in small displays)
- swapping the checkbox with add/remove buttons
- removing short name field from available columns
- applying the filter to the list of available columns

Screen Shot 2019-04-05 at 11 54 42 AM

Screen Shot 2019-04-05 at 11 55 17 AM

Screen Shot 2019-04-05 at 1 08 23 PM

## Testing

- checkout this pr
- navigate to {PROJECT_ROOT}/metron-deployment/development/centos6
- start full dev wm (more in the Metron documentation)
- navigate to {PROJECT_ROOT}/metron-interface/metron-alerts/
- run ./scripts/start-dev.sh
- make sure the functionality described abowe works fine

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel).
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

**Comment by TonyJabbour** [comment]
hello @tiborm can i have please your contact i have couple of questions regarding metron

**Comment by tiborm** [comment]
> hello @tiborm can i have please your contact i have couple of questions regarding metron

Hi Tony, I suggest to reach out to the community to make sure you find the right person you looking for. You can find more on this [here](http://metron.apache.org/community/).

**Comment by TonyJabbour** [comment]
Alright thank you, i went to you as i saw you are active, anw thanks

File: metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.spec.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ import { DialogService } from 'app/service/dialog.service';
 import { Observable } from 'rxjs';
 import { Filter } from 'app/model/filter';
 
-fdescribe('AlertsListComponent', () => {
+describe('AlertsListComponent', () => {
```
Comment (ruffle1986) [inline metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.spec.ts]: You should not fix this in the PR. It's fixed in https://github.com/apache/metron/pull/1372 (there's another fdescribe in utils btw)

File: metron-interface/metron-alerts/src/app/alerts/configure-table/configure-table.component.ts [hunk 2]
```diff
@@ -165,21 +168,18 @@ export class ConfigureTableComponent implements OnInit, AfterViewInit {
...
-      this.saveColumnNames();
-    });
-
-
+    this.configureTableService.saveColumnMetaData(
+      this.visibleColumns.map(columnMetaWrapper => columnMetaWrapper.columnMetadata))
+      .subscribe(() => {
+        this.saveColumnNames();
+      }, error => {
+        console.log('Unable to save column preferences ...');
```
Comment (ruffle1986) [inline metron-interface/metron-alerts/src/app/alerts/configure-table/configure-table.component.ts]: shouldn't it be removed?

File: metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.spec.ts (hunk 1 above)
Comment (tiborm) [reply metron-interface/metron-alerts/src/app/alerts/alerts-list/alerts-list.component.spec.ts]: I should remove this to let all tests running in Travis. Don't worry about the conflict, we'll resolve it if it's needed.

File: metron-interface/metron-alerts/src/app/alerts/configure-table/configure-table.component.ts (hunk 2 above)
Comment (tiborm) [reply metron-interface/metron-alerts/src/app/alerts/configure-table/configure-table.component.ts]: I haven't touched this part of the code, only a few lines above. By theory, we could remove console.log statements or bind them to isDevMode(), but hence console.logs are everywhere in the code I think we just make it inconsistent.

**Comment by mmiklavc** [comment]
@tiborm - I tested this feature out in full dev and I like the improvements! This makes column selection much, much better imo. I have only one question before I give this a +1. I took a look at the `user_settings` table in HBase after modifying the layout, saving it, and then logging out and back in again. The configured view still remains intact, however the settings table is empty. What are we actually storing there, if anything? Is the user's view only cached in a client-side cookie, or some other location?

```
# echo "list" | hbase shell
HBase Shell; enter 'help<RETURN>' for list of supported commands.
Type "exit<RETURN>" to leave the HBase Shell
Version 1.1.2.2.6.5.1050-37, r897822d4dd5956ca186974c10382e9094683fa29, Tue Dec 11 02:04:10 UTC 2018

list
TABLE
enrichment
enrichment_list
metron_update
profiler
threatintel
user_settings
6 row(s) in 0.4670 seconds

["enrichment", "enrichment_list", "metron_update", "profiler", "threatintel", "user_settings"]
[root@node1(127.0.0.1 192.168.66.121): /usr/metron/0.7.1]
# echo "scan 'user_settings'" | hbase shell
HBase Shell; enter 'help<RETURN>' for list of supported commands.
Type "exit<RETURN>" to leave the HBase Shell
Version 1.1.2.2.6.5.1050-37, r897822d4dd5956ca186974c10382e9094683fa29, Tue Dec 11 02:04:10 UTC 2018

scan 'user_settings'
ROW                           COLUMN+CELL
0 row(s) in 0.3840 seconds
```

**Comment by tiborm** [comment]
@mmiklavc Column names and the layout is stored in the browser's local storage. I think in the long run it would be better to replace local storage with a small user settings specific microservice.

**Comment by mmiklavc** [comment]
Confirming my +1 stands. Probably worth noting that we have a user_settings table, but don't appear to be using it.

**Pull request closed without merging**